File: python/container-with-most-water.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        highest = 0
        l = 0
        r = len(heights) - 1

        while l < r:
            area = min(heights[l],heights[r]) * (r-l)
            highest = max(highest, area)

            if heights[l] < heights[r]:
                l += 1
            else:
                r -= 1

        return highest

File: javascript/easy/container-with-most-water.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let maxWater = 0;
        let l = 0
        let r = heights.length - 1;

        while (l < r){
            let len = r - l;
            let area = Math.min(heights[r],heights[l]) * len;
            maxWater = Math.max(maxWater,area);

            if(heights[l] < heights[r]) l++;
            else r--;
        }
        return maxWater;
    }
}